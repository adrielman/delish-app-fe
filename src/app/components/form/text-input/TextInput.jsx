import React from "react";
import "./TextInput.scss";

const TextInput = props => {
  const onChange = event => {
    const { name, value } = event.target;
    props.onChange(name, value);
  };

  return (
    <div className="TextInput">
      <input
        placeholder={props.label}
        value={props.value}
        name={props.name}
        onChange={onChange}
      />
    </div>
  );
};

export default TextInput;
