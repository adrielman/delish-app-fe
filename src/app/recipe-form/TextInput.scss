@import "./../../styles/shared.scss";

.TextInput {
  display: flex;
}
