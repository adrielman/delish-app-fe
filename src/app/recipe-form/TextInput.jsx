import React, { Component } from "react";
import "./TextInput.scss";

const TextInput = () => {
  const onChange = event => {
    const { name, value } = event.target;
    this.props.onChange(name, value);
  };

  return (
    <div className="TextInput">
      <input
        placeholder={this.props.label}
        value={this.props.value}
        name={this.props.name}
        onChange={this.onChange}
      />
    </div>
  );
};

export default TextInput;
