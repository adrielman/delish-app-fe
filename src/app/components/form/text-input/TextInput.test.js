import React from 'react';
import { shallow } from 'enzyme';
import { TextInput } from './TextInput';

describe('TextInput', () => {
  const props = {
    label: 'Label',
    value: 'Value',
    name: 'Name',
    onChange: jest.fn()
  };
  const component = shallow(<TextInput {...props} />);

  it('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('should call onChange when value is changed', () => {
    component
      .find('input')
      .simulate('change', { target: { name: 'Name', value: 'Changed value' } });
    expect(props.onChange).toBeCalledTimes(1);
    expect(props.onChange).toBeCalledWith('Name', 'Changed value');
  });
});
