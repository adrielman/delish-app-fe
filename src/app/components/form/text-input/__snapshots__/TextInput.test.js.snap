// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should match snapshot 1`] = `
<div
  className="TextInput"
>
  <input
    name="Name"
    onChange={[Function]}
    placeholder="Label"
    value="Value"
  />
</div>
`;
